contracts/adapters/FinancingChainlink.sol

```diff
@@ -0,0 +1,185 @@
+     * @param token The token to receive the funds.
+     * @param amount The desired amount.
+     * @param data Additional details about the financing proposal.
+     */
+    function submitProposal(
+        DaoRegistry dao,
+        bytes32 proposalId,
+        address applicant,
+        address token,
+        uint256 amount,
+        bytes memory data
+    ) external override reentrancyGuard(dao) {
+        require(amount > 0, "invalid requested amount");
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        require(bank.isTokenAllowed(token), "token not allowed");
+        require(
+            isNotReservedAddress(applicant),
+            "applicant using reserved address"
+        );
+
+        dao.submitProposal(proposalId);
+
+        ProposalDetails storage proposal = proposals[address(dao)][proposalId];
+        proposal.applicant = applicant;
+        proposal.amount = amount;
+        proposal.token = token;
+
+        IVoting votingContract = IVoting(dao.getAdapterAddress(VOTING));
+        address sponsoredBy =
+            votingContract.getSenderAddress(
+                dao,
+                address(this),
+                data,
+                msg.sender
+            );
+
+        dao.sponsorProposal(proposalId, sponsoredBy, address(votingContract));
+        votingContract.startNewVotingForProposal(dao, proposalId, data);
+    }
+
+    /**
+     * @notice Processing a financing proposal to grant the requested funds.
+     * @dev Only proposals that were not processed are accepted.
+     * @dev Only proposals that were sponsored are accepted.
+     * @dev Only proposals that passed can get processed and have the funds released.
+     * @param dao The DAO Address.
+     * @param proposalId The proposal id.
+     */
+    function processProposal(DaoRegistry dao, bytes32 proposalId)
+        external
+        override
+        reentrancyGuard(dao)
+    {
+        ProposalDetails memory details = proposals[address(dao)][proposalId];
+
+        IVoting votingContract = IVoting(dao.votingAdapter(proposalId));
+        require(address(votingContract) != address(0), "adapter not found");
+
+        require(
+            votingContract.voteResult(dao, proposalId) ==
+                IVoting.VotingState.PASS,
+            "proposal needs to pass"
+        );
+        dao.processProposal(proposalId);
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+
+        //enter US Dollar = details.amount to convert it into ETH
+        uint256 amount = _usdToEth(details.amount);
+
+        bank.subtractFromBalance(GUILD, details.token, amount);
+        bank.addToBalance(details.applicant, details.token, amount);
+    }
+
+    /**
+     *  Returns the latest price from AggregatorV3Interface.sol
+        Chainlink returns the latest price to 5 decimials 
+        e.g. 176471000000 = $1,764.71
+     */
+    function getLatestPrice() public view returns (int256) {
+        (
```

Comment:
if you only need answer back, you can leave every other variable names blank. This uses less gas since it just ignore those values and does not put them in the stack

Reply:
Even though we do not use them, I think these variable names are needed, because the AggregatorInterfaceV3 from Chainlink looks for all these variables.